commit 8e7167ad0ea919e3af46eebdd1ccdaba3dd93acc (refs/changes/64/4764/4)
Author: Mike Gerdts <mike.gerdts@joyent.com>
Date:   2018-08-29T20:25:18+00:00 (1 year, 1 month ago)
    
    OS-7145 lx audit misses bounds check
    Reviewed by: Patrick Mooney <patrick.mooney@joyent.com>
    Reviewed by: Jerry Jelinek <jerry.jelinek@joyent.com>
    Approved by: Jerry Jelinek <jerry.jelinek@joyent.com>

diff --git a/usr/src/uts/common/brand/lx/os/lx_audit.c b/usr/src/uts/common/brand/lx/os/lx_audit.c
index de68bcafb1..6e522e6d8d 100644
--- a/usr/src/uts/common/brand/lx/os/lx_audit.c
+++ b/usr/src/uts/common/brand/lx/os/lx_audit.c
@@ -862,6 +862,9 @@ lx_audit_syscall_exit(int sysnum, long ret)
 	if (lxzd->lxzd_audit_enabled == LXAE_DISABLED)
 		return;
 
+	if (sysnum >= LX_NSYSCALLS)
+		return;
+
 	asp = lxzd->lxzd_audit_state;
 	ASSERT(asp != NULL);
 
